### DISCUSSION TITLE
Prevent new REST API creation when upgrading from version 1 to 2

### DISCUSSION THREAD

**Discussion by vdiezel** [serverless/serverless#11097] [Q&A] [answered]
Hey!

First of all: Thank you for the package!

Second: I have a problem.

After dragging it out for way too long, we would like to upgrade our serverless package to version 3. We are on version 1 right now and I attempted to upgrade to version 2 first. Our services are mainly HTTP endpoints - so it's just a bunch of Lambdas and REST API.s We are using the http-Event for Lambdas, so our APIs are REST APIs in AWS API Gateway. We are using custom domains attached via the serverless-domain-manager plugin. We are **not** using the HTTP API (httpAPI).

When deploying one of our HTTP services with version 2, I've realized that the following happens:
-  the first deployment fails, saying that deleting the API failed because a custom domain was still attached to it

![Screenshot from 2022-05-20 11-17-22](https://user-images.githubusercontent.com/37934155/169496922-0bc5c827-4463-4534-a1dd-f4a7f88f7a2e.png)

-  running the deployment a second time worked, but it created a new API next to the old one. The old API has no HTTP endpoints anymore, but the custom domain is still attached to it. The new API had all the HTTP endpoints and no custom domain.

This is problematic for us because the new API IDs are breaking access policies, and the missing custom domains will make the endpoints unreachable, until we have applied post-deployment fixes. Before going down that road, I'd like to know if I'm doing something wrong. 

So my question is: 

**Is the creation of a new REST API intended when going from v1 (1.79.0) to v2? If so, is there a way to prevent that? And if not, what am I doing wrong?** 

  Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              14.15.0
     Framework Version:         2.72.3 (local)
     Plugin Version:            5.5.4
     SDK Version:               4.3.2
     Components Version:        3.18.2

**Comment by vdiezel** [comment, marked as answer]
I'll answer myself: 

The import-plugin https://www.serverless.com/plugins/serverless-import-config-plugin doesn't work on version 2.72.1 (probably also below that). Not a big deal as it can be replaced with the `${file()}` syntax. 
That means our service was deployed without resources first, i.e. without our HTTP endpoints. Once we had fixed the import, the HTTP endpoint were back which then causes the creation of a new REST API as the reference to the old one is gone. 

By fixing the imports before the first deployment, one can circumvent this problem.

I'm glad we discovered that problem on our HTTP endpoints and not our user pools.

**Reply by pgrzesik** [reply]
Thanks a lot for sharing the solution, it might help someone in the future 🙌
